### ISSUE TITLE
Cat_boost encoder 

### ISSUE THREAD

**Issue by xuanqing94** [scikit-learn-contrib/category_encoders#311]
https://github.com/scikit-learn-contrib/category_encoders/blob/a810a4b7abfce9fc4eb7fc401e3d37f2c1c6e402/category_encoders/cat_boost.py#L278

That does't look like an interpolation of mean of global vs. mean of subgroup. I think it should be something like:

```python
 level_means = ((colmap['sum'] + self._mean * self.a) / (colmap['count'] + self.a)).where(level_notunique, self._mean) 
```

**Comment by GLevV** [comment]
Yep, [this commit](https://github.com/scikit-learn-contrib/category_encoders/commit/25accdc9b5b1998ae317e2bcd120464a7b1027bf) forget to add multiplication by a. It's a bug, good catch.

[Reference papaer, eq 1](http://learningsys.org/nips17/assets/papers/paper_11.pdf)

**Comment by PaulWestenthanner** [comment]
fixed by #339 
Thanks :)
